### PR TITLE
Alerting: Fix updating alert rule properties with missing/zero values

### PR DESCRIPTION
**Pull request by papagian**
**What this PR does / why we need it**:
Fix deleting labels and annotations: https://github.com/grafana/grafana/issues/34628#issuecomment-849560404
I have preserved behaviour and if title, condition, ~interval, for~, queries and expressions are missing then they are copied from the previous rule version.
I've extended the above by copying also no data and error state if they are missing.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by wbrowne — APPROVED** [review]
LGTM

**Comment by papagian** [comment]
@wbrowne thanks for approving it but I have made some more changes and added some tests. Would you mind to check once more, thanks!

File: pkg/tests/api/alerting/api_alertmanager_test.go [hunk 1]
```diff
@@ -1395,6 +1397,230 @@ func TestAlertRuleCRUD(t *testing.T) {
...
+					  "exec_err_state":"Alerting"
+				       }
+				    }
+				 ]
+			      }
+			   ]
+			}`, body)
+	}
+
+	// update the rule; keep title, condition, interval, for, queries and expressions if not provided
```
Comment (wbrowne) [inline pkg/tests/api/alerting/api_alertmanager_test.go:1523]: Is this case meant to actually update anything? IE Are `NoDataState` + `ExecErrState` meant to be the same as before?

Comment (papagian) [reply pkg/tests/api/alerting/api_alertmanager_test.go]: good question; I don't think that the UI will ever send empty values for those states but I have change the behaviour and keep them as well.

**Review by wbrowne — APPROVED** [review]

**Pull request merged**